Implemented first unit test

lordscales91 opened the pull request:
Well, I have managed to implement the first unit test. It's a test for the `CreateMMDModelRoot` operator. I have described there in the comments the usage for that test but this is the general rule:

`blender [blender args...] --python <script file> -- [script arguments...]` 
- `--background` is used to execute Blender without the UI
- `-noaudio` is used to disable audio in Blender
- `--verbose` is passed to the test script and tells [unittest](https://docs.python.org/3/library/unittest.html#cmdoption-unittest-discover-v) to give a verbose output

I was about to implement tests for the rest of the model operators. But those would require some sample data and I was unsure about what should be the best approach to distribute those samples among developers. 

The easiest approach would be to just put them in the repository but. Repositories should only contain source code if possible. So, my idea is to upload the samples to some server (Dropbox, Onedrive, Google Drive...) and then place on the repository the download links. The drawback of this option is that the download links may be unavailable and automatize the download can be troublesome. What do you think?

Additional note: It is not necessary to use unittest for the test scripts, they could be just simple scripts that process some data and check the results, but unittest should be preferred.

lordscales91 committed:
Implemented first unit test

powroupi commented on tests/test_model_operators.py:
```diff
@@ -0,0 +1,61 @@
+# outside the cp1252 range. Because Blender parses the file with that character set.
+# http://stackoverflow.com/questions/3284827
+# https://developer.blender.org/T35176
+
```
Could it be the issue that you did not add `# -*- coding: utf-8 -*-` at first line? Because `u'全ての親'` and `u'表情'` just work fine in our scripts.

lordscales91 replied:
I tried adding that line, with the same result, seems it is ignored. The process of [loading addons](https://developer.blender.org/diffusion/B/browse/master/tests/python/bl_load_addons.py) is different from executing a script from the command line. The [addon_utils](https://developer.blender.org/diffusion/B/browse/master/release/scripts/modules/addon_utils.py) python module imports found modules and calls its `register()` method. However, when executing a script from the command line, the file is loaded into memory and then the code is compiled and executed in the internal Python interpreter. So, when loading the file presumably Blender uses the system default encoding and ignores the `coding` header.

powroupi replied:
Understood. It's kind of a limitation..., we may imporve this in the future if there is a way to solve it. :)

powroupi commented:
> I was about to implement tests for the rest of the model operators. But those would require some sample data and I was unsure about what should be the best approach to distribute those samples among developers.

Could it be possible to setup test scene in script so that we may not need to use external sample data?
For the pmx or pmd files, maybe we can just put the links of some free shared model, so we download and put them in a specific directory manually, then we can start the tests. And we can just show `'specific model file needed to do the tests'` or similar messages if the data not being found. Would this idea be better/easier?

lordscales91 commented:
> Could it be possible to setup test scene in script so that we may not need to use external sample data?

It depends on which part of the code we are testing. Material morphs can be tested with a simple cube, the same is true for UV morphs. However the build/clean rig operators rely on a physics setup, which may be difficult to generate programmatically.

> For the pmx or pmd files, maybe we can just put the links of some free shared model, so we download and put them in a specific directory manually

Yes, ideally it would be fine to automate the download process, but of course the models could be downloaded manually. That way we don't need to implement code logic to download the files. It could be more difficult to find free models for this purpose, given the draconian rules that most of them have HA HA HA. But I hope this will not be a problem

powroupi commented:
> It could be more difficult to find free models for this purpose, given the draconian rules that most of them have HA HA HA. But I hope this will not be a problem

I think pmd files might not be a problem because we can use the default models which are included in mmd program. Others maybe we can find some from [VPVP wiki](http://www6.atwiki.jp/vpvpwiki/pages/141.html) or [MikuMikuDance_Wiki](http://mikumikudance.wikia.com/wiki/MikuMikuDance_Wiki). About the rules, i don't think we will break the rules because we just use them to test our addon locally.  :confused:

For now, we can just do the 'rule free' things and imporve other things step by step in the future. :smile: 

And please let me know when you completed your changes, i'll merge it soon. :)

lordscales91 committed:
Replaced Base64 strings for unicode literals on tests

lordscales91 commented:
Well, I replaced the Base64 strings for Unicode literals which would be more efficient. I managed to get the literal string sequence encoding the string with 'unicode-escape' codec. I executed it on the Blender's Interactive console

![Interactive Console](http://i.imgur.com/hHJegSO.png)

The good think about that console is that is not really a shell, it's part of the Blender UI, which allows to show unicode characters even on Windows. Then on my IDE I just replaced the `\\u` for `\u`. It's a bit troublesome, but is the best we have for now. Maybe the Ubuntu Bash that will come with the [anniversary update](http://www.howtogeek.com/248177/whats-new-in-windows-10s-anniversary-update/) will allow unicode characters, but I'm afraid it will not be the case.

You can merge it now. I just wanted to provide a sample unit test to help other developers get started to create unit tests for the addon. I will also contribute with more tests when I have time, but for now this is enough.

powroupi commented:
Thank you for your changes. It's a good start point! :smile:
